lib/testing.dart

```diff
@@ -0,0 +1,133 @@
+import 'dart:async';
+
+import 'package:web_http/web_http.dart';
+
+/// A signature for callback which is invoked whenever the [TestHttpClient]
+/// recieves a request.
+typedef FutureOr<TestHttpResponse> TestClientCallback(String url, String method,
```

Comment:
Consider using the new typedef syntax? 

```dart
typedef RequestHandler = FutureOr<TestHttpResonse> Function(
  String url,
  String method, {
  String responseType,
  String body,
  Map<String, String> headers,
  int timeout,
  bool withCredentials,
});
```

Reply:
That unfortunately doesn't allow naming url and method.  I could also make everything named, but I want to keep it synced with the request method.

Reply:
Oh really? That almost seems like a bug. :confused: